### PR TITLE
gowin: Add support for MUX2_LUT5/6/7/8

### PR DESCRIPTION
**Pull request by yrabbit**
Hardware MUXes do not have separate inputs so LUTs are always used, with some slight optimization.

### CODE REVIEW

File: techlibs/gowin/synth_gowin.cc [hunk 1]
```diff
@@ -112,30 +111,28 @@ struct SynthGowinPass : public ScriptPass
...
+			if (args[argidx] == "-vout" && argidx + 1 < args.size()) {
 				vout_file = args[++argidx];
 				continue;
 			}
-			if (args[argidx] == "-json" && argidx+1 < args.size()) {
+			if (args[argidx] == "-json" && argidx + 1 < args.size()) {
 				json_file = args[++argidx];
 				nobram = true;
 				nolutram = true;
-				nowidelut = true;
```
Comment (pepijndevos) [inline techlibs/gowin/synth_gowin.cc:129]: Took me a while to find the actual key change between all the formatting changes haha

File: techlibs/gowin/synth_gowin.cc [hunk 2]
```diff
@@ -17,17 +17,16 @@
...
+#include "kernel/register.h"
+#include "kernel/rtlil.h"
 
 USING_YOSYS_NAMESPACE
 PRIVATE_NAMESPACE_BEGIN
 
-struct SynthGowinPass : public ScriptPass
-{
-	SynthGowinPass() : ScriptPass("synth_gowin", "synthesis for Gowin FPGAs") { }
+struct SynthGowinPass : public ScriptPass {
```
[the remark this answers is not in the document]
Comment (pepijndevos) [reply techlibs/gowin/synth_gowin.cc:28]: Apparently there is git-clang-format that does this. sweet https://twitter.com/KenoFischer/status/1447240880658784257

[the remark this answers is not in the document]
Comment (yrabbit) [reply techlibs/gowin/synth_gowin.cc]: cleaned:)

**Comment by yrabbit** [comment]
oops. missed it :)

**Pull request closed without merging**